### PR TITLE
[Documentation:Developer] Virtual Machine Snapshots

### PR DESCRIPTION
**Pull request by musaabimran**
In the event of a non-recoverable error while working on Submitty, the last resort is a time-consuming vagrant up. To save time, take a snapshot during the initial Vagrant setup, allowing for quick recovery by reverting to the saved state if needed. Updated documentation as this might help developers.

### CODE REVIEW

File: _docs/developer/development_instructions/index.md [hunk 1]
```diff
@@ -197,6 +197,23 @@ autograding configuration, you'll probably need to:
 
 * [Batch Regrade Homeworks](/instructor/batch_regrade_submissions) already submitted to those gradeables.
 
+---
+
+## Virtual Machine Recovery using Snapshots
+
+In the event of a non-recoverable error while working on Submitty the last resort is to perform a fresh `vagrant up`. However, this process can be time-consuming. To avoid such situations and save time, it is highly recommended to take a snapshot when you first set up your Vagrant environment by following the tutorial links provided below:
```
Comment (NiaHeermance) [inline _docs/developer/development_instructions/index.md:204]: Should be "...while working on Submitty, the last resort is to..." (add a comma)

**Review by NiaHeermance — APPROVED** [review]
The failed Ruby test will be fixed when Satvik's PR is merged, so you're all good there. This documentation is great! I added a comment on a run-on sentence fix, but otherwise you're all good.

File: _docs/developer/development_instructions/index.md (hunk 1 above)
Comment (musaabimran) [reply _docs/developer/development_instructions/index.md]: Updated it!

**Review by ziesski — COMMENTED** [review]
Tested on Mac, and works :)

**Pull request merged**